cmd/cody-gateway/internal/httpapi/completions/fireworks.go

```diff
@@ -211,10 +212,20 @@ func (f *FireworksHandlerMethods) parseResponseAndUsage(logger log.Logger, reqBo
 		if len(event.Choices) > 0 {
 			completionUsage.characters += len(event.Choices[0].Text)
 		}
+		// these are only included in the last message, so we're not worried about overwriting
+		if event.Usage.PromptTokens > 0 {
+			promptUsage.tokens = event.Usage.PromptTokens
+		}
+		if event.Usage.CompletionTokens > 0 {
+			completionUsage.tokens = event.Usage.CompletionTokens
+		}
 	}
 	if err := dec.Err(); err != nil {
 		logger.Error("failed to decode Fireworks streaming response", log.Error(err))
 	}
+	if completionUsage.tokens <= 0 || promptUsage.tokens <= 0 {
```

Comment:
We set these to -1 as sentinel values in L193

```suggestion
	if completionUsage.tokens == -1 || promptUsage.tokens == -1 {
```

Reply:
Fair, I initially thought we can override the `-1` with zeros, but the way the code is written now it's either -1 or >0